Dockerfile-linux.template

```diff
@@ -285,6 +286,11 @@ RUN set -eux; \
 -}}
 	; \
 	\
+{{ if is_alpine and (.version | version_id) < ("8" | version_id) then ( -}}
+# Replace musl iconv headers with gnu, only necessary for PHP < 8
+	cp /usr/include/gnu-libiconv/*.h /usr/include; \
```

Comment:
I think we should try to be a tiny bit more proactive about making sure musl's iconv doesn't get used; maybe something like this:

```suggestion
{{ if is_alpine then ( -}}
# make sure musl's iconv doesn't get used (https://www.php.net/manual/en/intro.iconv.php)
	rm -vf /usr/include/iconv.h; \
{{ if (.version | version_id) < ("8" | version_id) then ( -}}
# PHP < 8 doesn't know to look deeper for GNU libiconv: https://github.com/php/php-src/commit/b480e6841ecd5317faa136647a2b8253a4c2d0df
	ln -sv /usr/include/gnu-libiconv/*.h /usr/include/; \
{{ ) else "" end -}}
```